src/python/pants/help/scope_info_iterator.py

```diff
@@ -55,14 +61,14 @@ def _expand_tasks(self, scopes):
                     outer = enclosing_scope(outer)
         return sorted(expanded_scopes)
 
-    def _expand_subsystems(self, scope_infos):
+    def _expand_subsystems(self, scope_infos: List[ScopeInfo]) -> Iterator[ScopeInfo]:
```

Comment:
Ditto on if it's safe for this to be `Iterable[ScopeInfo]`.

Reply:
At this point the code in `ScopeInfoIterator` has started to impose a notion of order on the outputs, so this does have to be an order-imposing `List`

Reply:
Technically, if all you care about is stable insertion order, it's better to use `Sequene[ScopeInfo]` so that you allow things like tuples and `OrderedSet`. For a function parameter, you only want to require `List` if you truly need list features like `.append`.